app/src/main/java/com/example/homework3/fragments/ListFragment.kt

```diff
@@ -0,0 +1,102 @@
+package com.example.homework3.fragments
+
+import android.graphics.Rect
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.homework3.adapter.UserAdapter
+import com.example.homework3.databinding.FragmentListBinding
+import com.example.homework3.retrofit.GithubService
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+class ListFragment : Fragment() {
+    private var _binding: FragmentListBinding? = null
+    private val binding: FragmentListBinding
+        get() = requireNotNull(_binding) {
+            "View was destroyed"
+        }
+    private val viewModel: MVViewModel by viewModels {
+        object : ViewModelProvider.Factory {
+            override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+                @Suppress("UNCHECKED_CAST")
+                return MVViewModel(GithubService.provideApi()) as T
+            }
+        }
+    }
+
+    private val adapter = UserAdapter { user ->
+        findNavController().navigate(
+            ListFragmentDirections.toDetails(
+                user.login
+            )
+        )
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return FragmentListBinding.inflate(inflater, container, false)
+            .also { binding ->
+                _binding = binding
+            }.root
+    }
+
+    /**
+     * я так понимаю что если используется viewLifecycleOwner
+     * то закрывать поток не надо закрывать потому что он работает на
+     * жизненном цикле view и при ее уничтожении он закрывается автоматом
+     */
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(binding) {
+            val linearLayoutManager = LinearLayoutManager(view.context)
+            recyclerView.adapter = adapter
+            recyclerView.layoutManager = linearLayoutManager
```

Comment:
можно LinearLayoutManager объявить через xml
```suggestion
            recyclerView.adapter = adapter
            recyclerView.layoutManager = LinearLayoutManager(view.context)
```